ibis-server/app/model/data_source.py

```diff
@@ -163,6 +163,6 @@ def get_trino_connection(
             port=int(info.port.get_secret_value()),
             database=info.catalog.get_secret_value(),
             schema=info.trino_schema.get_secret_value(),
-            user=info.user.get_secret_value(),
-            password=info.password.get_secret_value(),
+            user=(info.user and info.user.get_secret_value()),
+            password=(info.password and info.password.get_secret_value()),
```

Comment:
Curiously, why need to have an `and` operation here? For null checking? If yes, should we do the same thing for other properties?

Reply:
Yes, for null. Other information about the data source is not allowed to be null.